src/test/test-capability.c

```diff
@@ -77,7 +84,39 @@ static int setup_tests(void) {
         test_uid = nobody->pw_uid;
         test_gid = nobody->pw_gid;
 
-        return 0;
+        /* Find out the kernel version to determine if the ambient
+         * capability tests should be run. */
+
+        if (uname(&u) < 0)
+                return -errno;
+
+        version = strdup(u.release);
+        orig = version;
+
+        token = strsep(&version, ".");
+        if (!token || !version) {
+                r = -EINVAL;
+                goto finish;
+        }
+
+        r = safe_atou(token, &major);
+        if (r < 0)
+                goto finish;
+
+        /* Can't use safe_atou here because the string is not clean. */
+        minor = strtoul(version, &end, 10);
+        if (version == end) {
+                r = -EINVAL;
+                goto finish;
+        }
+
+        if (major > 4 || (major == 4 && minor >= 3))
+                *run_ambient = true;
+
+        r = 0;
+finish:
+        free(orig);
```

Comment:
Please use `_cleanup_free_` for `orig`. That will make the error paths a lot more readable.